src/draft-wullink-rpp-requirements.md

```diff
@@ -78,122 +78,124 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT","SHOULD", "SH
 
 **R1.1.** A wel defined architecture MUST be defined for RPP, including a description of the responsibilities of the definded protocol layers. 
 
-**R1.2.** The API MUST provide a clear, clean, easy to use and self-explanatory interface that can easily be integrated into existing software systems and includes language bindings for the most popular programming languages.
-//PK: too vague
-//see: https://github.com/ietf-wg-rpp/rpp-requirements/issues/3
+**R1.2.** The API MUST provide a clear, intuitive, and self-explanatory interface that facilitates seamless integration into existing software systems. Identifiers used in the API MUST avoid patterns, that could conflict with the syntax or semantics of most popular programming languages.
+> //TODO: [Issue #3](https://github.com/ietf-wg-rpp/rpp-requirements/issues/3)
 
-**R1.3.** RPP MUST leverage widely deployed web standards, tools, and infrastructure components such as HTTP, JSON, [!@OpenAPI] specification, API gateways, load balancing, caching and delegate functional responsibility to the HTTP layer when possible. For example, authentication is part of the HTTP layer and not part of the RPP application layer.
-//PK:  too vague
-//see: https://github.com/ietf-wg-rpp/rpp-requirements/issues/4
+**R1.3.** Wherever applicable RPP MUST leverage existing best practices and well adopted standards for building and documenting RESTful APIs
```

Comment:
Should this language perhaps try and align more with how R2.2 is worded? Theses two seem to me close in spirit so IMHO they should try and be as close in wording as possible as well.

Reply:
R1.3 is a general (or meta) requirement, whereas R2.2 is a reference to/implementation of it in context of HTTP where we have a clear guidance from BCP.

I notice however that here we have a MUST and in R2.2 there is SHOULD, which is likely the contradiction you are alluding to.

I reopened #4 to take care of it and will move your comment in there as well.